src/executor-group.jl

```diff
@@ -178,15 +182,31 @@ function forward(self:: DataParallelExecutorGroup, data_provider :: AbstractData
    # TODO add callbacks here
 end
 
+# TODO Add description
+backward(self::DataParallelExecutorGroup, out_grads::Void) = backward(self, NDArray[])
```

Comment:
Ah that makes sense. Then we should update the module API definition to make a requirement for all `<: AbstractModule`.